x-pack/plugins/apm/public/components/app/service_overview/service_overview_charts/latency_map/embedded_map.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, useRef } from 'react';
+import uuid from 'uuid';
+import {
+  MapEmbeddable,
+  MapEmbeddableInput,
+  MapEmbeddableOutput,
+} from '@kbn/maps-plugin/public';
+import { MAP_SAVED_OBJECT_TYPE } from '@kbn/maps-plugin/common';
+import {
+  ErrorEmbeddable,
+  ViewMode,
+  isErrorEmbeddable,
+  EmbeddableFactory,
+} from '@kbn/embeddable-plugin/public';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import { EuiText } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+import { ApmPluginStartDeps } from '../../../../../plugin';
+import { getLayerList } from './get_layer_list';
+import { useApmParams } from '../../../../../hooks/use_apm_params';
+import { useTimeRange } from '../../../../../hooks/use_time_range';
+
+export function EmbeddedMapComponent({ filters }: { filters: Filter[] }) {
+  const {
+    query: { rangeFrom, rangeTo, kuery },
+  } = useApmParams('/services/{serviceName}/overview');
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const [error, setError] = useState<boolean>();
+
+  const [embeddable, setEmbeddable] = useState<
+    MapEmbeddable | ErrorEmbeddable | undefined
+  >();
+
+  const embeddableRoot: React.RefObject<HTMLDivElement> =
+    useRef<HTMLDivElement>(null);
+
+  const {
+    embeddable: embeddablePlugin,
+    maps,
+    notifications,
+  } = useKibana<ApmPluginStartDeps>().services;
+
+  useEffect(() => {
+    async function setupEmbeddable() {
+      const factory:
+        | EmbeddableFactory<
+            MapEmbeddableInput,
+            MapEmbeddableOutput,
+            MapEmbeddable
+          >
```

Comment:
Shouldn't this type be inferred by TS based on the response of `embeddablePlugin?.getEmbeddableFactory`?

Reply:
It's not inferred, but I've moved it to the call of `getEmbeddableFactory`